engine/src/main/java/com/agiletec/aps/system/common/AbstractGenericCacheWrapper.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2018-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package com.agiletec.aps.system.common;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.entando.entando.aps.system.exception.CacheItemNotFoundException;
+import org.springframework.cache.Cache;
+
+/**
+ * @author E.Santoboni
+ * @param <O> The object to manage
+ */
+public abstract class AbstractGenericCacheWrapper<O extends Object> extends AbstractCacheWrapper {
+
+	protected static enum Action {
+		ADD, UPDATE, DELETE
+	}
+
+	protected void releaseCachedObjects(Cache cache) {
+		List<String> codes = (List<String>) this.get(cache, this.getCodesCacheKey(), List.class);

```

Comment:
I think we should add some trace or debug logging at the top level to all of these calls. What do you think?